codegen-test/model/misc.smithy

```diff
@@ -108,3 +113,69 @@ union AUnion {
     string: String,
     time: Timestamp,
 }
+
+/// This operation tests that the response code defaults to 200 when no other code is set
+@httpResponseTests([
+    {
+        id: "ResponseCodeDefaultTest",
+        protocol: "aws.protocols#restJson1",
+        code: 200,
+    }
+])
+@http(method: "GET", uri: "/responseCodeDefault")
+operation ResponseCodeDefault {
+    input: ResponseCodeDefaultInput,
+    output: ResponseCodeDefaultOutput,
+}
+
+@input
+structure ResponseCodeDefaultInput {}
+
+@output
+structure ResponseCodeDefaultOutput {}
+
+/// This operation tests that the response code defaults to @http's code
+@httpResponseTests([
+    {
+        id: "ResponseCodeHttpFallbackTest",
+        protocol: "aws.protocols#restJson1",
+        code: 418,
+    }
+])
+@http(method: "GET", uri: "/responseCodeHttpFallback", code: 418)
+operation ResponseCodeHttpFallback {
+    input: ResponseCodeHttpFallbackInput,
+    output: ResponseCodeHttpFallbackOutput,
+}
+
+@input
+structure ResponseCodeHttpFallbackInput {}
+
+@output
+structure ResponseCodeHttpFallbackOutput {}
+
+/// This operation tests that @httpResponseCode is @required
+/// and is used over @http's code
```

Comment:
Not a good test, because `http.code` is `418`, so you wouldn't know if it's using `params.responseCode` or falling back to `http.code`.